components/layout/fragment.rs

```diff
@@ -877,11 +878,32 @@ impl Fragment {
 
     /// Adds a style to the inline context for this fragment. If the inline
     /// context doesn't exist yet, it will be created.
-    pub fn add_inline_context_style(&mut self, style: Arc<ComputedValues>) {
+    pub fn add_inline_context_style(&mut self,
+                                    style: Arc<ComputedValues>,
+                                    first_frag: bool,
+                                    last_frag: bool) {
+
+        let mut frag_style : Arc<ComputedValues>;
         if self.inline_context.is_none() {
             self.inline_context = Some(InlineFragmentContext::new());
+        }        
+        if first_frag && last_frag {
```

Comment:
Could just be `let frag_style = if first_frag && last_frag { ... } else { ... }` and avoid having to make `frag_style` mutable.